File: app/src/main/java/com/cna/mineru/cna/Fragment/HomeFragment.java
package com.cna.mineru.cna.Fragment;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Handler;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.GridView;

import com.cna.mineru.cna.AddNote;
import com.cna.mineru.cna.DB.GraphSQLClass;
import com.cna.mineru.cna.DB.HomeSQLClass;
import com.cna.mineru.cna.DB.ImageSQLClass;
import com.cna.mineru.cna.DB.UserSQLClass;
import com.cna.mineru.cna.DTO.HomeData;
import com.cna.mineru.cna.Adapter.GridAdapter;
import com.cna.mineru.cna.ModifyHomeItem;
import com.cna.mineru.cna.R;
import com.cna.mineru.cna.Utils.DefaultInputDialog;
import com.cna.mineru.cna.Utils.LoadingDialog;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.ArrayList;


public class HomeFragment extends Fragment {

    private HomeSQLClass db;
    private GraphSQLClass gp_db;
    private ImageSQLClass i_db;
    private UserSQLClass u_db;

    private GridView gv;
    private GridAdapter mAdapater;
    private FloatingActionButton fb;
    private LoadingDialog loadingDialog;

    private int note_id = 0;

    private ArrayList<HomeData> list  = new ArrayList<>();

    public HomeFragment(){

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_home, container, false);
        fb = (FloatingActionButton)view.findViewById(R.id.fb_add);
        db = new HomeSQLClass(getActivity());
        u_db= new UserSQLClass(getActivity());
        gp_db = new GraphSQLClass(getActivity());
        i_db = new ImageSQLClass((getActivity()));
        loadingDialog = new LoadingDialog();
        mAdapater = new GridAdapter(getContext(), R.layout.row, list);

        list = db.load_values();

        gv = (GridView)view.findViewById(R.id.gridView1);
        gv.setAdapter(mAdapater);

        fb.setOnClickListener(new View.OnClickListener(){
            @Override
            public void  onClick(View v){
                if(!u_db.isClassChecked()){
                    DefaultInputDialog d = new DefaultInputDialog();
                    d.show(getActivity().getSupportFragmentManager(),"setting");
                    d.setDialogResult(new DefaultInputDialog.OnMyDialogResult() {
                        @Override
                        public void finish(int _class) {
                            DefaultInputDialog d2 = new DefaultInputDialog();
                            d2.show(getActivity().getSupportFragmentManager(),"setting2");
                            d2.setDialogResult(new DefaultInputDialog.OnMyDialogResult() {
                                @Override
                                public void finish(int _class) {

                                }
                            });
                        }
                    });
                }else{
                    list.clear();
                    fb.setEnabled(false);
                    Handler h = new Handler();
                    h.postDelayed(new splashHandler(), 1000);
                    Intent i =new Intent(getActivity(),AddNote.class);
                    startActivity(i);
                }
            }
        });

        gv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                loadingDialog.progressON(getActivity(),"Loading...");
                gv.setEnabled(false);
                Handler h = new Handler();
                h.postDelayed(new splashHandler(), 1000);

                HomeData data;
                data = db.select_item(list.get(position).id);
                Intent i = new Intent(getActivity(),ModifyHomeItem.class);
                i.putExtra("id", data.id);
                i.putExtra("title", data.title_text);
                i.putExtra("tag", data.tag);
                startActivity(i);
                getActivity().overridePendingTransition(R.anim.anim_slide_in_left, R.anim.anim_slide_out_right);
            }
        });

        gv.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
            @Override
            public boolean onItemLongClick(AdapterView<?> parent, View view, final int position, long id) {
                try{
                    AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
                    builder.setTitle("오답노트 삭제");
                    builder.setMessage("정말로 삭제하시겠습니까?");
                    builder.setPositiveButton("예",
                            new DialogInterface.OnClickListener() {
                                public void onClick(DialogInterface dialog, int which) {
                                    db.delete_item(list.get(position).id);
                                    gp_db.delete_value(list.get(position).id);
                                    i_db.delete_item(list.get(position).id);
                                    note_id = list.get(position).id;
                                    new DelNote().execute(getString(R.string.ip_set)+"/api/note/destroy");
                                    onResume();
                                }
                            });
                    builder.setNegativeButton("아니오",
                            new DialogInterface.OnClickListener() {
                                public void onClick(DialogInterface dialog, int which) {
                                }
                            });
                    builder.show();

                }
                catch (Exception e) {
                }
                return true;
            }
        });
        return view;
    }

    @SuppressLint("StaticFieldLeak")
    public class DelNote extends AsyncTask<String, String, String> {
        @Override
        protected String doInBackground(String... urls) {
            try {
                JSONObject jsonObject = new JSONObject();
                jsonObject.accumulate("id", note_id);
                HttpURLConnection con = null;
                BufferedReader reader = null;
                URL url = new URL(urls[0]);
                try {
                    con = (HttpURLConnection) url.openConnection();
                    con.setRequestMethod("DELETE");//DELETE 방식으로 보냄
                    con.setRequestProperty("Connection", "Keep-Alive");
                    con.setRequestProperty("Cache-Control", "no-cache");//캐시 설정
                    con.setRequestProperty("Content-Type", "application/json");//application JSON 형식으로 전송
                    con.setRequestProperty("Accept", "text/html");//서버에 response 데이터를 html로 받음
                    con.setRequestProperty("Accept-Charset", "UTF-8");
                    con.setDoOutput(true);//Outstream으로 PUT 데이터를 넘겨주겠다는 의미
                    con.setDoInput(true);//Inputstream으로 서버로부터 응답을 받겠다는 의미
                    con.connect();

                    OutputStream outStream = con.getOutputStream();
                    BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(outStream));
                    writer.write(jsonObject.toString());
                    writer.flush();
                    writer.close();

                    InputStreamReader stream = new InputStreamReader(con.getInputStream(), "UTF-8");

                    reader = new BufferedReader(stream);

                    StringBuffer buffer = new StringBuffer();

                    String line = "";

                    while ((line = reader.readLine()) != null) {//(중요)서버로부터 한줄씩 읽어서 문자가 없을때까지 넣어줌
                        buffer.append(line + "\n"); //읽어준 스트링값을 더해준다.
                    }
                    line = buffer.toString();
                    return line;//서버로 부터 받은 값을 리턴해줌 아마 OK!!가 들어올것임
                } catch (IOException e) {
                    e.printStackTrace();
                } finally {
                    if (con != null) {
                        con.disconnect();
                    }
                    try {
                        if (reader != null) {
                            reader.close();
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(String result) {
            super.onPostExecute(result);
            int error = 0;
            JSONObject jObject = null;
            try {
                jObject = new JSONObject(result);
                error = jObject.optInt("error");
                if(error==2){

                }else {

                }

            } catch (JSONException e) {
                e.printStackTrace();
            }
            loadingDialog.progressOFF();
        }
    }

    private class splashHandler implements Runnable{
        public void run()	{
            fb.setEnabled(true); // 클릭 유효화
            gv.setEnabled(true);

        }
    }

    @Override
    public void onResume() {
        super.onResume();
        list = db.load_values();
        mAdapater = new GridAdapter(getContext(), R.layout.row, list);
        gv.setAdapter(mAdapater);
    }

    @Override
    public void onStop(){
        super.onStop();
        loadingDialog.progressOFF();
    }
}

File: app/src/main/java/com/cna/mineru/cna/Fragment/HomeFragment.kt
package com.cna.mineru.cna.Fragment

import android.annotation.SuppressLint
import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.GridView
import androidx.fragment.app.Fragment
import com.google.android.material.floatingactionbutton.FloatingActionButton

import com.cna.mineru.cna.Adapter.GridAdapter
import com.cna.mineru.cna.AddNote
import com.cna.mineru.cna.DB.GraphSQLClass
import com.cna.mineru.cna.DB.HomeSQLClass
import com.cna.mineru.cna.DB.ImageSQLClass
import com.cna.mineru.cna.DB.UserSQLClass
import com.cna.mineru.cna.DTO.HomeData
import com.cna.mineru.cna.ModifyHomeItem
import com.cna.mineru.cna.R
import com.cna.mineru.cna.Utils.CustomDialog
import com.cna.mineru.cna.Utils.DefaultInputDialog
import com.cna.mineru.cna.Utils.LoadingDialog

import org.json.JSONException
import org.json.JSONObject

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.IOException
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.net.HttpURLConnection
import java.net.URL
import java.util.ArrayList

import kotlinx.android.synthetic.main.fragment_home.*
class HomeFragment : Fragment() {

    private var db: HomeSQLClass? = null
    private var gp_db: GraphSQLClass? = null
    private var i_db: ImageSQLClass? = null
    private var u_db: UserSQLClass? = null

    private var gv: GridView? = null
    private var mAdapater: GridAdapter? = null
    private var fb: FloatingActionButton? = null
    private var loadingDialog: LoadingDialog? = null

    private var note_id = 0

    private var list = ArrayList<HomeData>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        val view = inflater?.inflate(R.layout.fragment_home, container, false)
        fb = view?.findViewById(R.id.fb_add) as FloatingActionButton
        db = HomeSQLClass(activity)
        u_db = UserSQLClass(activity)
        gp_db = GraphSQLClass(activity)
        i_db = ImageSQLClass(activity)
        loadingDialog = LoadingDialog()
        mAdapater = GridAdapter(context, R.layout.home_item, list)

        list = db!!.load_values()

        gv = view.findViewById(R.id.gridView1)
        gv!!.adapter = mAdapater

        fb!!.setOnClickListener {
            if (!u_db!!.isClassChecked) {
                val d = DefaultInputDialog()
                activity?.supportFragmentManager?.let { it1 -> d.show(it1, "setting") }
                d.setDialogResult(object : DefaultInputDialog.OnMyDialogResult {
                    override fun finish(_class: Int) {
                        val d2 = DefaultInputDialog()
                        activity?.supportFragmentManager?.let { it1 -> d2.show(it1, "setting2") }
                        d2.setDialogResult(object : DefaultInputDialog.OnMyDialogResult {
                            override fun finish(_class: Int) {

                            }
                        })
                    }
                })
            } else {
                list.clear()
                fb!!.isEnabled = false
                val h = Handler()
                h.postDelayed(splashHandler(), 1000)
                val i = Intent(activity, AddNote::class.java)
                startActivity(i)
            }
        }

        gv!!.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            loadingDialog!!.progressON(activity, "Loading...")
            gv!!.isEnabled = false
            val h = Handler()
            h.postDelayed(splashHandler(), 1000)

            val data: HomeData
            data = db!!.select_item(list[position].id)
            val i = Intent(getActivity(), ModifyHomeItem::class.java)
            i.putExtra("id", data.id)
            i.putExtra("title", data.title_text)
            i.putExtra("tag", data.Tag)
            i.putExtra("subtag", data.Subtag)
            i.putExtra("isCalledHome", true)
            startActivity(i)
            activity?.overridePendingTransition(
                R.anim.anim_slide_in_left,
                R.anim.anim_slide_out_right
            )
        }

        gv!!.onItemLongClickListener =
            AdapterView.OnItemLongClickListener { parent, view, position, id ->
                try {
                    val d = CustomDialog(6)
                    activity?.supportFragmentManager?.let { d.show(it, "alter") }
                    d.setDialogResult(object : CustomDialog.OnMyDialogResult {
                        override fun finish(_class: Int) {
                            if (_class == 1) {
                                db!!.delete_item(list[position].id)
                                gp_db!!.delete_value(list[position].id)
                                i_db!!.delete_item(list[position].id)
                                note_id = list[position].id
                                //                                new DelNote().execute(getString(R.string.ip_set)+"/api/note/destroy");
                                onResume()
                            } else
                                d.dismiss()
                        }

                        override fun finish(result: Int, email: String) {

                        }
                    })
                } catch (ignored: Exception) {
                }

                true
            }
        return view
    }

    @Suppress("UNREACHABLE_CODE")
    @SuppressLint("StaticFieldLeak")
    inner class DelNote : AsyncTask<String, String, String>() {
        override fun doInBackground(vararg urls: String): String? {
            try {
                val jsonObject = JSONObject()
                jsonObject.accumulate("id", note_id)
                var con: HttpURLConnection? = null
                var reader: BufferedReader? = null
                val url = URL(urls[0])
                try {
                    con = url.openConnection() as HttpURLConnection
                    con.requestMethod = "DELETE"//DELETE 방식으로 보냄
                    con.setRequestProperty("Connection", "Keep-Alive")
                    con.setRequestProperty("Cache-Control", "no-cache")//캐시 설정
                    con.setRequestProperty(
                        "Content-Type",
                        "application/json"
                    )//application JSON 형식으로 전송
                    con.setRequestProperty("Accept", "text/html")//서버에 response 데이터를 html로 받음
                    con.setRequestProperty("Accept-Charset", "UTF-8")
                    con.doOutput = true//Outstream으로 PUT 데이터를 넘겨주겠다는 의미
                    con.doInput = true//Inputstream으로 서버로부터 응답을 받겠다는 의미
                    con.connect()

                    val outStream = con.outputStream
                    val writer = BufferedWriter(OutputStreamWriter(outStream))
                    writer.write(jsonObject.toString())
                    writer.flush()
                    writer.close()

                    val stream = InputStreamReader(con.inputStream, "UTF-8")

                    reader = BufferedReader(stream)

                    val buffer = StringBuffer()

                    var line = ""
                    //(중요)서버로부터 한줄씩 읽어서 문자가 없을때까지 넣어줌
                    do{
                        line = reader.readLine()
                        buffer.append(line + "\n")
                        //읽어준 스트링값을 더해준다.
                    }while (true)
                    line = buffer.toString()
                    return line//서버로 부터 받은 값을 리턴해줌 아마 OK!!가 들어올것임
                } catch (e: IOException) {
                    e.printStackTrace()
                } finally {
                    con?.disconnect()
                    try {
                        reader?.close()
                    } catch (e: IOException) {
                        e.printStackTrace()
                    }

                }
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return null
        }

        override fun onPostExecute(result: String) {
            super.onPostExecute(result)
            var error = 0
            var jObject: JSONObject? = null
            try {
                jObject = JSONObject(result)
                error = jObject.optInt("error")
                if (error == 2) {

                } else {

                }

            } catch (e: JSONException) {
                e.printStackTrace()
            }

            loadingDialog!!.progressOFF()
        }
    }

    private inner class splashHandler : Runnable {
        override fun run() {
            fb!!.setEnabled(true) // 클릭 유효화
            gv!!.isEnabled = true

        }
    }

    override fun onResume() {
        super.onResume()
        list = db!!.load_values()
        mAdapater = GridAdapter(getContext(), R.layout.home_item, list)
        gv!!.adapter = mAdapater
    }

    override fun onStop() {
        super.onStop()
        loadingDialog!!.progressOFF()
    }
}
